alf/algorithms/containers.py

```diff
@@ -0,0 +1,502 @@
+            state=(alg_step.state, echo_output),
+            info=alg_step.info)
+
+    def rollout_step(self, inputs, state):
+        block_state, echo_state = state
+        block_input = dict(input=inputs, echo=echo_state)
+        alg_step = self._alg.rollout_step(block_input, block_state)
+        real_output = alg_step.output['output']
+        echo_output = alg_step.output['echo']
+        return AlgStep(
+            output=real_output,
+            state=(alg_step.state, echo_output),
+            info=alg_step.info)
+
+    def train_step(self, inputs, state, rollout_info):
+        block_state, echo_state = state
+        block_input = dict(input=inputs, echo=echo_state)
+        alg_step = self._alg.train_step(block_input, block_state, rollout_info)
+        real_output = alg_step.output['output']
+        echo_output = alg_step.output['echo']
+        return AlgStep(
+            output=real_output,
+            state=(alg_step.state, echo_output),
+            info=alg_step.info)
+
+    def calc_loss(self, info):
+        return self._alg.calc_loss(info)
+
+    def preprocess_experience(self, root_inputs, rollout_info, batch_info):
+        return self._alg.preprocess_experience(root_inputs, rollout_info,
+                                               batch_info)
+
+    def after_update(self, root_inputs, info):
+        self._alg.after_update(root_inputs, info)
+
+    def after_train_iter(self, root_inputs, rollout_info):
+        self._alg.after_train_iter(root_inputs, rollout_info)
+
+
+@alf.configurable
+class RLAlgWrapper(RLAlgorithm):
+    """Wrap an Algorithm as RLAlgorithm so that it can be used for RLTrainer"""
+
+    def __init__(self,
+                 observation_spec,
+                 action_spec,
+                 algorithm,
+                 env=None,
+                 reward_spec=alf.TensorSpec(()),
+                 reward_weights=None,
+                 config: TrainerConfig = None,
+                 optimizer=None,
+                 debug_summaries=False,
+                 name="RLAlgWrapper"):
+        super().__init__(
+            observation_spec=observation_spec,
+            action_spec=action_spec,
+            train_state_spec=algorithm.train_state_spec,
+            reward_spec=reward_spec,
+            predict_state_spec=algorithm.predict_state_spec,
+            rollout_state_spec=algorithm.rollout_state_spec,
+            reward_weights=reward_weights,
+            is_on_policy=algorithm.on_policy,
+            env=env,
+            config=config,
+            optimizer=optimizer,
+            debug_summaries=debug_summaries,
+            name=name)
+
+        self._algorithm = algorithm
+
+    def set_on_policy(self, is_on_policy):
+        super().set_on_policy(is_on_policy)
+        self._algorithm.set_on_policy(is_on_policy)
+
+    def set_path(self, path):
+        super().set_path(path)
+        self._alg.set_path(path)
+
+    def rollout_step(self, inputs: TimeStep, state):
```

Comment:
Yeah I mean instead of calling self._algorithm.xxx() one by one, we can just use one function to forward the call? For example:

```python
def __getattr__(self, name):
    if name in ["rollout_step", "train_step", ...]:
          return getattr(self._algorithm, name)
```